docs/contribute/refactoring-filesystem.md

```diff
@@ -0,0 +1,105 @@
+
+1.  It maps a reference points object to a Saros project object internally 
+2. It provides the functionalities of the Saros project
+
+Because a reference point itself has no methods, it is important that the CoreReferencePointManager provides functionalities of a Saros project, so that co-existance of both architecture is possible.
+
+
+### Saros/E
+
+In situations, like processing activities, Saros/E converts Saros resources backs to Eclipse resource. So it is very useful, that the mapper in Saros/E gets the features, that it determinates Eclipse resources given by reference points and relative paths to the resources. This mapper is called EclipseReferencePointManager and has following features:
+
+1.  It maps a reference point object to an Eclipse project internally
+2.  It determines Eclipse resources in combination of reference points and relative paths to the resources
+
+### Saros/I
+
+The IntelliJReferencePointManager is needed for Saros/I and has two responsibilities:
+
+1.  It maps a reference point object to an IntelliJ Module internally
+2.  It determines java resources in combination of reference points and relative paths to the resources
+
+### Saros Server
+
+The ServerReferencePointManager is needed for Saros Server and has two responsibilities:
+
+1.  It maps a reference point object to an Java IO file resource internally
+2.  It determines java resources in combination of reference points and relative paths to the resources
+
+All ReferencePointManager are not needed only during the refactoring. There are also needed especially handling Saros activities (ex. A saros resources was created). All activities contains a Saros path (SPath), which consists of a Saros project and a relative path to the resource, which the activity triggered. The Spath has additionally feature: It also returns the affected Saros resources.
+
+With the refactoring to reference points, the ReferencePointManager can be very helpful handling activities, especially determining the IDE depended resources.
+
+### Imported Refactoring Steps
+
+For introducing the reference point based interfaces, but also keeping project-based interfaces during the refactoring, follwing steps are useful:
+
+1.  Mark project-based interface as deprecated
+2.  Insert the reference point based interface
+3.  Implement the reference point based interface
+4.  Remove the project-based interface's implementation and reference this to the reference point based interface
+5.  Find alle usages of project-based interfaces and reference them to reference point based interface
+6.  Delete project based interface
+
+This procedure has the adventages, that the parallel development of Saros is provided, so that developer can use the reference point based interfaces and the (still) untouched components are not affected directly. After all usages references the reference point based interface, the project based one can be deleted.
```

Comment:
```suggestion
This procedure has the advantages, that the parallel development of Saros is provided so that developers can use the reference point based interfaces and the (still) untouched components are not affected directly. After the reference point based interfaces are fully adopted, the project-based interfaces can be deleted.
```